src/Autoloader.php

```diff
@@ -116,17 +116,14 @@ public static function getPsr4()
     /**
      *
      */
-    public static function addClassMap(array $classMap)
+    public static function addClassMap(array $classMap, $path)
     {
         // If not registered, register.
         if (!static::isRegistered()) {
             static::register();
         }
-
-        static::$classMap = array_merge(
-            static::$classMap,
-            $classMap
-        );
+        $path = rtrim($path, DIRECTORY_SEPARATOR);
+        static::$classMap[$path] = $classMap;
```

Comment:
I have thinkg to this.
But I have huge doubt it will make sense to have multiple "classmap" in the same folder.